pac4j-core/src/main/java/org/pac4j/core/profile/service/InMemoryProfileService.java

```diff
@@ -0,0 +1,90 @@
+package org.pac4j.core.profile.service;
+
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.definition.CommonProfileDefinition;
+import org.pac4j.core.util.CommonHelper;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * In-memory profile service.
+ *
+ * @author Elie Roux
+ * @since 2.0.0
+ */
+
+public class InMemoryProfileService<U extends CommonProfile> extends AbstractProfileService<U>  {
+
+	public Map<String,Map<String,Object>> profiles;
+	public Function<Object[], U> profileFactory;
+
+	public InMemoryProfileService(final Function<Object[], U> profileFactory) {
```

Comment:
I'm not sure but would it make sense to provide also a parameter-less constructor with a default profile factory for `CommonProfile`?

Actually, I would even:
 - remove any `profileFactory` from the constructor
 - call defaultProfileDefinition in the `internalInit()` method with a stupid `CommonProfileDefinition` initialised using an anonymous `CommonProfile` factory.
 - let the user call setProfileDefinition() if changing it is needed.

WDYT?